src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -873,3 +873,70 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
 
   return TRUE;
 }
+
+/**
+ * ostree_repo_static_delta_verify_signature:
+ * @superblock: Static-delta superblock GVariant
+ * @sign: Signature engine used to check superblock
+ * @error: Error
+ *
+ * Verify the superblock data, except inline commits, usin @sign engine.
+ */
+gboolean
+ostree_repo_static_delta_verify_signature (GVariant         *superblock,
+                                           OstreeSign       *sign,
+                                           GError          **error)
+{
+  gsize size = g_variant_type_n_items(G_VARIANT_TYPE (OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT));
+  g_autoptr(GVariantBuilder) desc_sign_builder = NULL;
+  g_autoptr(GVariant) delta_meta = g_variant_get_child_value (superblock, 0);
+
+  /* Check if there are signatures in the superblock for signature engine */
+  const gchar *signature_key = ostree_sign_metadata_key(sign);
+  GVariantType *signature_format = (GVariantType *) ostree_sign_metadata_format(sign);
+  g_autoptr(GVariant) signatures = g_variant_lookup_value (delta_meta,
+                                                           signature_key,
+                                                           signature_format);
+  if (!signatures)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "no signature for '%s' in static-delta superblock",
+                   signature_key);
+      return FALSE;
```

Comment:
why return `FALSE` in case if the superblock do not use the signing?

Reply:
This tests if there is a signature for a specific 'sign' engine, passing another engine should fail

Reply:
correct -- didn't check the purpose of the function.